ui/src/app/groupings/[selectedGrouping]/layout.tsx

```diff
@@ -0,0 +1,35 @@
+import ExportDropdown from './_components/export-dropdown';
+import GroupingHeader from './_components/grouping-header';
+import ReturnButtons from './_components/return-buttons';
+import SideNav from './_components/side-nav';
+import {groupingDescription} from '@/lib/fetchers';
+
+const SelectedGroupingLayout = async ({ params, tab }: { params: { selectedGrouping: string }; tab: React.ReactNode }) => {
+    const res = await groupingDescription(params.selectedGrouping);
+    const groupPath = res?.groupPath || '';
+    const groupDescription = res?.description || '';
+    const groupName = groupPath.split(':').pop() || '';
```

Comment:
Sorry about this, everything in fetchers.ts and actions.ts should have `& ApiError` removed. For example:
```
export const groupingDescription = async (groupingPath: string): Promise<GroupingDescription & ApiError> => {
```

Then you can reformat the code like so:
```
const { description } = await groupingDescription(params.selectedGrouping);
```
Also you use the groupingPath from the params prop instead of from the response. We want to guarantee the groupingPath displays even if groupingDescription() throws an error.